src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -751,3 +902,135 @@ RouterExecutorEnd(QueryDesc *queryDesc)
 	queryDesc->estate = NULL;
 	queryDesc->totaltime = NULL;
 }
+
+
+/* TODO: write comment */
+void
+InstallRouterExecutorShmemHook()
+{
+	prev_shmem_startup_hook = shmem_startup_hook;
+	shmem_startup_hook = RegisterRouterExecutorXactCallbacks;
+}
+
+
+/* TODO: Write comment */
+static void
+RegisterRouterExecutorXactCallbacks()
+{
+	RegisterXactCallback(RouterTransactionCallback, NULL);
```

Comment:
> the startup hooks get called on each backend startup (since there's no fork to copy state)

Isn't that what I want here? I need these to run on every backend, which is why they're in the `shmem_startup_hook`, which I thought ran during backend initialization. Did I do this incorrectly?